src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/jobs/NotifyAttentionNeededLicencesServiceTest.kt

```diff
@@ -0,0 +1,209 @@
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.writeCsv
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+class NotifyAttentionNeededLicencesServiceTest {
+  private val licenceRepository = mock<LicenceRepository>()
+  private val prisonerSearchApiClient = mock<PrisonerSearchApiClient>()
+  private val notifyService = mock<NotifyService>()
+
+  private val service = NotifyAttentionNeededLicencesService(
+    emailAddress,
+    licenceRepository,
+    prisonerSearchApiClient,
+    notifyService,
+  )
+
+  @BeforeEach
+  fun reset() {
+    val authentication = mock<Authentication>()
+    val securityContext = mock<SecurityContext>()
+
+    whenever(authentication.name).thenReturn("smills")
+    whenever(securityContext.authentication).thenReturn(authentication)
+    setContext(securityContext)
+
+    reset(
+      licenceRepository,
+      prisonerSearchApiClient,
+      notifyService,
+    )
+  }
+
+  @Test
+  fun `notify attention needed licences job should return if there are no licences in attention needed tab`() {
+    whenever(licenceRepository.getAttentionNeededLicences()).thenReturn(listOf())
+
+    service.notifyAttentionNeededLicences()
+
+    verify(prisonerSearchApiClient, times(0)).searchPrisonersByNomisIds(emptyList())
+    verify(notifyService, times(0)).sendAttentionNeededLicencesEmail(emailAddress, ByteArray(0), fileName)
+  }
+
+  @Test
+  fun `notify attention needed licences job should send email if there are licences in attention needed tab`() {
+    whenever(licenceRepository.getAttentionNeededLicences()).thenReturn(listOf(aLicenceEntity))
+    whenever(prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(aLicenceEntity.nomsId.toString()))).thenReturn(
+      listOf(aPrisonerSearchPrisoner),
+    )
+
+    service.notifyAttentionNeededLicences()
+    val fileContent = writeCsv(
+      listOf(
+        NotifyAttentionNeededLicence(
+          aLicenceEntity.nomsId,
+          aPrisonerSearchPrisoner.prisonName,
+          aPrisonerSearchPrisoner.legalStatus,
+          aLicenceEntity.conditionalReleaseDate,
+          aLicenceEntity.actualReleaseDate,
+          aLicenceEntity.licenceStartDate,
+        ),
+      ),
+    )
+    val cvsData =
+      "Noms ID,Prison Name,Noms Legal Status,ARD,CRD,Licence Start Date\r\nA1234AA,null,SENTENCED,2021-10-22,2021-10-22,2021-10-22\r\n"
+
+    assertThat(fileContent).isEqualTo(cvsData)
+    verify(prisonerSearchApiClient, times(1)).searchPrisonersByNomisIds(listOf(aLicenceEntity.nomsId.toString()))
+    verify(notifyService, times(1)).sendAttentionNeededLicencesEmail(
+      emailAddress,
+      fileContent.toByteArray(),
+      fileName,
+    )
+  }
+
+  private companion object {
+    val emailAddress = "testemail@probation.gov.uk"
+    val fileName = "attentionNeededLicences_" + LocalDate.now() + ".csv"
+    val aLicenceEntity = TestData.createCrdLicence().copy(
```

Comment:
The benefit of using copy is that you only need to provide values for relevant fields - this is nice as it draws attention to what attributes are important for this functionality. I would suggest trimming it down to the smallest point that the tests still pass